skorch/dataset.py

```diff
@@ -69,6 +70,11 @@ def _len(x):
 
 
 def get_len(data):
+    if hasattr(data, 'convert_to_tensors'):
+        # this might be expensive but huggingface BatchEncodings are lists of
+        # lists and thus their length would be determined incorrectly otherwise,
+        # returning the sequence length instead of the number of samples.
+        data = data.convert_to_tensors('pt')
```

Comment:
Since we are special casing for `BatchEncodings` anyways, can we have custom logic get the outer list length just for `BatchEncodings`?

Reply:
Yes, good idea, I changed the code not to do the conversion but instead just return the length of the outer list. Also, instead of checking if the `convert_to_tensors` method exists, I now check if we deal with a `Mapping` that has the `input_ids` key.